docs/articles/game-engines/article_01.md

```diff
@@ -0,0 +1,189 @@
+responsive machine will be a great aid along the way.
+
+Once again, you will need Windows. For our Mac friends out there, the harsh truth
+is that both the game industry and Apple are working against you when it comes to
+game development. For now, we will only focus on Windows. If you have an Intel-based
+Mac, set up a dual boot machine with Windows.
+
+I would highly recommend at least 8GB of RAM, 16GB if you plan to use an IDE or
+an IDE-like text editor such as Visual Studio Code.
+
+### Software Prerequisites
+
+You will need a small list of software to get started.
+
+* [Microsoft Visual Studio](https://visualstudio.microsoft.com/downloads/)
+
+    Visual Studio contains the MSVC toolchain which will allow us to compile C/C++
+    code to our machine. Once installed, it will launch Visual Studio Installer,
+    an update manager / plugin suite where you will select **Desktop Development
+    with C++**. This will ensure that MSVC is included with your installation of
+    Visual Studio. You can add other packages if you want, but we only need that one.
+
+* [CMake](https://cmake.org/download/)
+
+    We will also want CMake. CMake is a set of tools that makes building binary
+    projects easier. We will be using this to help generate our build files. In
+    our case, it will be Visual Studio solution files. This isn't a hard and
+    fast requirement; you can get away with just Visual Studio itself, but I highly
+    recommend that you use CMake instead since it decouples your source files from
+    the build tool (which would be MSVC). We won't be interacting much beyond writing
+    our initial CMake script and maintaining the list of source files we compile.
+
+    You will need to let CMake add itself to PATH for all users during installation
+    to allow it to work via CLI.
+
+* [Git](https://git-scm.com/)
+
+    If you don't already have Git installed, now is the time. You will want to use
+    an online source tracking service like Github to manage your project. Whether or
+    not it is public or private is up to you. This is a big project, so you want to
+    keep your code from disappearing if the rocks in your PC no longer conduct lightning.
+
+* Your Favorite Text Editor
+
+    If you use CMake like I told you, you won't need to use Microsoft Visual Studio
+    to write your code. I like Vim, some of you might like VS Code. Whatever environment
+    you are comfortable with. Use whatever. (Microsoft Word, anyone?)
+
+## Code Style, Over-Optimization, and Object-Orientated Programming/Design
+
+Before I send you off to write code, let's take a brief moment to talk about the act of
+actually writing code. For those that have talked to me in the past, then they might have heard me
+express some *opinions* about the current state of software development. Rather than
+peppering you with my ideas of how software *should be* throughout each and every
+article, I'll get it out of the way so there are no surprises later on.
```

Comment:
I actually do think that the reverse is more effective- It would be better to "show, don't tell" your style of programming throughout the course of the series and discuss your overall archetecture as it is built.

Again, it comes back to the credability issue. This site will (and already does) reach beyond just NIU, so the writers of any given article are more-or-less anonymous even if an author is named. Prefacing the series with "having some opinions about the current state of softdev" and other incendiary language comes off as a red flag when it is from someone with no prestige to back it up (no offense of course, it's just the nature of the site).

This also bleeds into the section regarding OOP & OOD. Without even debating the content of the section, it is problematic since it positions itself as an expert's perspective on a complex discussion. Since the GDL's main strength will be creating educational articles from the perspective of students and intermediates pooling their knowledge, it is somewhat inappropriate to "take sides" on higher level discussions like this.

Also, it will be difficult to convince game developers that OOP is ineffective up front, since it has been the backbone of many engines for a long while. Anyone who is experienced has already taken a side, and any newbies are going to eventually have to use it anyways, so the average impression will be neutral if not poor.

Cut the part about OOP and just demonstrate the effectiveness of your method through subsequent entries in the series. Since most people will probably not have experience in graphics programming, they won't notice a departure from OOP, and it would even actually be useful to point out non-OOP design patterns **after having shown them**. This let's you add to someone's toolset and pick which methods they like best, rather than try to dictate one way is bad and the other is good.

The advice about over-optimization and code style is good though, so keep that 👍

Reply:
This probably loops back to your previous point, and I do agree that "show, don't tell" might be a better point in illustrating some pitfalls of OOD. I do see your point though, my opinion on the matter really doesn't hold much weight unless I demonstrate it through practice.

I certainly injected some personality behind this too, so I'll probably trim most of this out and set up the expectation that I won't be utilizing OOP/OOD right out of the gate, but I will make a note to the readers that they can apply what I am doing to OOD should they so desire (which is entirely fine, hence the section about code style). This is one of the reasons why I added this section in.

Also, for the record, it isn't that I am against OOP/OOD, there are cases where OOP is highly appropriate (and I will be doing some for these tutorials), it's that I find it to be the leading cause for "refactor hell"; it takes a lot of skill to know when and where to use OOD for any given problem. But yeah, "the current state of the software development" was a bit of a hot take, I admit. lol

Reply:
No worries! Just wanting to make sure the site *adds* to discussions like these rather than declare one way or the other (unless something is objectively an anti-pattern, which I personally find is rarely the case). It's generally just a safer credability wise.